### ISSUE TITLE
`earliest_date` in a blueprint throws Carbon error

### ISSUE THREAD

**Issue by edalzell** [statamic/cms#4669]
### Bug description

In my date fields, in the blueprint, I have an `earliest_date` field that is set to `January 1, 1900`.

When I load that entry in the CP, I get a Carbon error
<img width="1137" alt="CleanShot 2021-11-05 at 13 36 44@2x" src="https://user-images.githubusercontent.com/6069653/140576172-207427fc-ed3c-4ec1-88f3-b8c9df5dc693.png">

If I remove that field from the blueprint it loads fine.

Note, I did not explicitly add this field, it was put in there, at some point, by Statamic when the field was added to the blueprint.

### How to reproduce

1. have a date field in a blueprint
2. add/verify that `earliest_date: 'January 1, 1900'` is in there
3. In the CP visit an entry that uses that blueprint

### Logs

```
[2021-11-05 20:57:12] local.ERROR: Unexpected data found.
Unexpected data found.
The separation symbol could not be found
Data missing {"userId":"cba0aed3-26f9-4ca9-a67f-8261a605d37b","exception":"[object] (Carbon\\Exceptions\\InvalidFormatException(code: 0): Unexpected data found.
Unexpected data found.
The separation symbol could not be found
Data missing at /Users/erin/Sites/transform/vendor/nesbot/carbon/src/Carbon/Traits/Creator.php:676)
[stacktrace]
#0 /Users/erin/Sites/transform/vendor/nesbot/carbon/src/Carbon/Traits/Creator.php(699): Carbon\\Carbon::rawCreateFromFormat('Y-m-d H:i', 'January 1, 1900', NULL)
#1 /Users/erin/Sites/transform/vendor/statamic/cms/src/Fieldtypes/Date.php(121): Carbon\\Carbon::createFromFormat('Y-m-d H:i', 'January 1, 1900')
#2 /Users/erin/Sites/transform/vendor/statamic/cms/src/Fieldtypes/Date.php(105): Statamic\\Fieldtypes\\Date->preProcessSingle('January 1, 1900')
#3 /Users/erin/Sites/transform/vendor/statamic/cms/src/Fields/Fieldtype.php(205): Statamic\\Fieldtypes\\Date->preProcess('January 1, 1900')
#4 /Users/erin/Sites/transform/vendor/statamic/cms/src/Fields/ConfigField.php(11): Statamic\\Fields\\Fieldtype->preProcessConfig('January 1, 1900')
#5 /Users/erin/Sites/transform/vendor/laravel/framework/src/Illuminate/Collections/HigherOrderCollectionProxy.php(60): Statamic\\Fields\\ConfigField->preProcess()
#6 [internal function]: Illuminate\\Support\\HigherOrderCollectionProxy->Illuminate\\Support\\{closure}(Object(Statamic\\Fields\\ConfigField), 'earliest_date')
#7 /Users/erin/Sites/transform/vendor/laravel/framework/src/Illuminate/Collections/Collection.php(664): array_map(Object(Closure), Array, Array)
#8 /Users/erin/Sites/transform/vendor/laravel/framework/src/Illuminate/Collections/HigherOrderCollectionProxy.php(61): Illuminate\\Support\\Collection->map(Object(Closure))
#9 /Users/erin/Sites/transform/vendor/statamic/cms/src/Fields/Fields.php(144): Illuminate\\Support\\HigherOrderCollectionProxy->__call('preProcess', Array)
#10 /Users/erin/Sites/transform/vendor/statamic/cms/src/Fields/Field.php(318): Statamic\\Fields\\Fields->preProcess()
#11 /Users/erin/Sites/transform/vendor/statamic/cms/src/Fields/Field.php(173): Statamic\\Fields\\Field->preProcessedConfig()
#12 /Users/erin/Sites/transform/vendor/laravel/framework/src/Illuminate/Collections/HigherOrderCollectionProxy.php(60): Statamic\\Fields\\Field->toPublishArray()
#13 [internal function]: Illuminate\\Support\\HigherOrderCollectionProxy->Illuminate\\Support\\{closure}(Object(Statamic\\Fields\\Field), 1)
#14 /Users/erin/Sites/transform/vendor/laravel/framework/src/Illuminate/Collections/Collection.php(664): array_map(Object(Closure), Array, Array)
#15 /Users/erin/Sites/transform/vendor/laravel/framework/src/Illuminate/Collections/HigherOrderCollectionProxy.php(61): Illuminate\\Support\\Collection->map(Object(Closure))
#16 /Users/erin/Sites/transform/vendor/statamic/cms/src/Fields/Fields.php(115): Illuminate\\Support\\HigherOrderCollectionProxy->__call('toPublishArray', Array)
#17 /Users/erin/Sites/transform/vendor/statamic/cms/src/Fields/Section.php(46): Statamic\\Fields\\Fields->toPublishArray()
```
### Versions

Statamic 3.2.20 Pro
Laravel 8.69.0
PHP 8.0.12
aryehraber/statamic-captcha 1.5.2
aryehraber/statamic-uuid 2.0.3
doublethreedigital/duplicator 2.0.1
edalzell/forma 1.0
jonassiewertsen/statamic-livewire 2.7.0
statamic/seo-pro 2.3.2
transformstudios/events 2.2.3
transformstudios/front 1.7.1
transformstudios/gated dev-main
transformstudios/magic-link dev-main
transformstudios/review 3.1
transformstudios/simple dev-main
transformstudios/stripe dev-main
transformstudios/takeover-redirects dev-main
transformstudios/uptime dev-main

### Installation

Starter Kit using via CLI

**Comment by duncanmcclean** [comment]
Saw this same thing yesterday, can confirm.

**Comment by jasonvarga** [comment]
Workaround is to remove `earliest_date` from your blueprint. You probably didn't want it there anyway.

**Comment by Ceepster14** [comment]
I have the same issue in a number of different collections attempting to both edit and create a new entry after upgrading to Statamic version 3.2.21

[2021-11-11 11:29:47] local.ERROR: Unexpected data found.
Unexpected data found.
The separation symbol could not be found
Data missing {"userId":"05a0515f-14ab-423b-b3bc-672f6b6b9b96","exception":"[object] (Carbon\\Exceptions\\InvalidFormatException(code: 0): Unexpected data found.
Unexpected data found.
The separation symbol could not be found

**Comment by Ceepster14** [comment]
Further to the last comment - if I try to edit the date field in the blueprint for one of the failing collections, it hangs and the Laravel log shows

[2021-11-11 11:43:46] local.ERROR: Unexpected data found.
Unexpected data found.
The separation symbol could not be found
Data missing {"userId":"05a0515f-14ab-423b-b3bc-672f6b6b9b96","exception":"[object] (Carbon\\Exceptions\\InvalidFormatException(code: 0): Unexpected data found.
Unexpected data found.
The separation symbol could not be found

**Comment by de-construct-mark** [comment]
Same here... after updating to 3.2.20

Carbon\Exceptions\InvalidFormatException
Unexpected data found. Unexpected data found. The separation symbol could not be found Data missing

**Comment by jonassiewertsen** [comment]
I can confirm the bug and that the workaround does fix it.

**Comment by jackmcdade** [comment]
Seems fixed in 3.3.
